static-site/src/components/Groups/edit-logo-form.jsx

```diff
@@ -0,0 +1,138 @@
+      const currentLogo = await getGroupLogo();
+      if (!cleanUp) setLogo({ ...logo, current: currentLogo });
+    }
+
+    let cleanUp = false;
+    setCurrentLogo();
+    return () => cleanUp = true;
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const getGroupLogo = async () => {
+    clearErrorMessage();
+    try {
+      const response = await fetch(`/groups/${groupName}/settings/logo`);
+      if (response.status === 404) return null;
+      if (response.ok) return URL.createObjectURL(await response.blob());
+      createErrorMessage(response.statusText);
+    } catch (err) {
+      console.error(err.message)
+      createErrorMessage();
+    }
+  }
+
+  const deleteGroupLogo = async (e) => {
+    e.preventDefault();
+    clearErrorMessage();
+    // TODO: Ask for confirmation before deleting?
+    setDeletionInProgress(true);
+
+    try {
+      const response = await fetch(`/groups/${groupName}/settings/logo`, {method: "DELETE"});
+      response.ok
+        ? setLogo({ ...logo, current: null })
+        : createErrorMessage(response.statusText);
+    } catch (err) {
+      console.error(err.message);
+      createErrorMessage()
+    }
+
+    setDeletionInProgress(false);
+  }
+
+  const uploadGroupLogo = async (e) => {
+    e.preventDefault();
+    clearErrorMessage();
+    setUploadInProgress(true);
+
+    try {
+      const response = await fetch(`/groups/${groupName}/settings/logo`, {
+        method: "PUT",
+        headers: {
+          "Content-Type": "image/png"
+        },
+        body: logo.new
+      });
+      response.ok
+        ? setLogo({ new: null, current: await getGroupLogo() })
+        : createErrorMessage(response.statusText);
+    } catch (err) {
+      console.error(err.message);
+      createErrorMessage();
+    }
+
+    setUploadInProgress(false);
+  }
+
+  const handleUploadButton = (e) => {
+    e.preventDefault();
+    clearErrorMessage();
+    fileInput.current.click();
+  }
+
+  const previewImage = (e) => {
+    e.preventDefault();
+    clearErrorMessage();
+    setLogo({ ...logo, new: e.target.files[0] });
+  }
+
+  return (
+    <CenteredForm>
+      {logo.current
+        ? <>
+            <splashStyles.H4>Current Logo:</splashStyles.H4>
+            <AvatarWithoutMargins alt="group-logo" src={logo.current}/>
+            <InputButton
+              disabled={deletionInProgress || uploadInProgress}
+              onClick={deleteGroupLogo}>
+              {deletionInProgress
+                ? "Deleting current logo..."
+                : "Delete current logo"
+              }
+            </InputButton>
+          </>
+        : <splashStyles.H4>No Current Logo</splashStyles.H4>
+      }
+
+      {/* Hide the file input to allow for custom text in button */}
+      <InputButton onClick={handleUploadButton}>
+        Choose new logo
+      </InputButton>
+      <input type="file" ref={fileInput} style={{ display: "none" }} accept="image/png" onChange={previewImage}/>
```

Comment:
You should be able to avoid the `fileInput` ref and the `handleUploadButton()` handler by wrapping your custom button content in a `<label>` for the `<input>`. Easiest way to do that is to make the input a child of the label:

```jsx
<label>
  <InputButton>Choose new logo</InputButton>
  <input type="file" style={{ display: "none" }} accept="image/png" onChange={previewImage}/>
</label>
```

but you can also use a separate id on the input and `<label for={id}>`.

Reply:
Hmm, using a `<label>` does display the button correctly, but the `<input>` click event does not get handled because `<InputButton>` is a styled button that has it's own click event. 

I could make a styled label for this specific button that shares the same CSS as `<InputButton>` to avoid the extra ref + event handler.

Reply:
Updated in e9f6caea09d943165b2fbb555012c6efcfdb19ce